src/fxos/readGeckoCommit.js

```diff
@@ -0,0 +1,97 @@
+let assert = require('assert');
+let debug = console.log.bind(console, '[fxos/readGeckoCommit]');
+let denodeify = require('denodeify');
+let fs = require('mz/fs');
+let path = require('path');
+let sax = require('sax');
+let tmpdir = denodeify(require('tmp').dir);
+
+const SOURCES_XML = '/system/sources.xml';
+const APPLICATION_INI = '/system/b2g/application.ini';
+const PLATFORM_INI = '/system/b2g/platform.ini';
+
+function pullIniFile(adb, local, file) {
+  return adb
+    .pull(file, local)
+    .then(() => path.join(local, path.basename(file)));
+}
+
+function readGeckoRevisionFromIni(iniFile) {
+  let revision;
+
+  return new Promise((resolve, reject) => {
+    let stream = fs.createReadStream(iniFile, {encoding: 'utf8'});
+
+    // The Gecko revision will be assigned to key named SourceStamp, in the
+    // line format:  SourceStamp=${REVISION}
+    stream.on('data', buffer => {
+      let line = buffer
+        .split('\n')
+        .find(line => line.includes('SourceStamp'));
+
+      if (!line) {
+        return;
+      }
+
+      revision = line.split('=')[1];
```

Comment:
Don't you want to remove the rejection listener here or check the revision against a regex or something?